rest/config_legacy.go

```diff
@@ -61,10 +61,12 @@ type LegacyServerConfig struct {
 	ReplicatorCompression                     *int                           `json:"replicator_compression,omitempty"` // BLIP data compression level (0-9)
 	BcryptCost                                int                            `json:"bcrypt_cost,omitempty"`            // bcrypt cost to use for password hashes - Default: bcrypt.DefaultCost
 	MetricsInterface                          *string                        `json:"metricsInterface,omitempty"`       // Interface to bind metrics to. If not set then metrics isn't accessible
+	DiagnosticInterface                       *string                        `json:"diagnosticInterface,omitempty"`    // Interface to bind diagnostic API to. If not set then diagnostic API isn't accessible
 	HideProductVersion                        bool                           `json:"hide_product_version,omitempty"`   // Determines whether product versions removed from Server headers and REST API responses. This setting does not apply to the Admin REST API.
 	DisablePersistentConfig                   *bool                          `json:"disable_persistent_config,omitempty" help:"Can be set to true to disable 3.0/persistent config handling."`
 	AdminInterfaceAuthentication              *bool                          `json:"admin_interface_authentication,omitempty" help:"Whether the admin API requires authentication"`
 	MetricsInterfaceAuthentication            *bool                          `json:"metrics_interface_authentication,omitempty" help:"Whether the metrics API requires authentication"`
+	DiagnosticInterfaceAuthentication         *bool                          `json:"diagnostic_interface_authentication,omitempty" help:"Whether the diagnostic API requires authentication"`
```

Comment:
I'm not set on supporting this in legacy config at all, but if we do:

https://github.com/couchbase/sync_gateway/blob/master/rest/config_legacy_test.go#L27 is where you should write a test for making sure that `diagnostic_interface_authentication` and `diagnosticInterface`.

I think a case could be made either way:

1. We don't want people using legacy config / we don't support it for scopes and collections, so we should not expose new features on it.
2. It's easy enough to support.